JobSeek/JobSeek.Api/Services/Implementation/BaseService.cs

```diff
@@ -15,49 +16,45 @@ public BaseService(IBaseRepository<TEntity> repository)
         {
             _repository = repository;
         }
-        public List<TEntity> GetAll()
-        {
-            return _repository.GetAll();
-        }
 
-        public List<TCustomEntity> GetAll<TCustomEntity>()
+        public virtual ListResponse<TCustomEntity> GetAll<TCustomEntity>()
             where TCustomEntity : BaseEntity
         {
-            return _repository.GetAll<TCustomEntity>();
+            var result = _repository.GetAll();
```

Comment:
you should use the _repository.GetAll<TEntity>() because you are working with TCustomEntity here not  TCustomEntity